meinberlin/apps/dashboard2/components.py

```diff
@@ -123,45 +197,77 @@ def _get_progress_for_object(object, required_fields):
 
 
 class ModuleFormComponent(ProjectFormComponent):
-    """Abstract interface for module dashboard components based on forms.
-
-    This component is intended to be used with ModuleDashboardForm.
-    It will always return a ModuleComponentFormView
-    and provides a default implementation for get_progress.
-
-    Required properties:
-    - menu_label: str
-      This label is always returned regardless of project state
-    - form_title: str
-      This title is shown on top of the rendered form
-    - form_class: ModuleDashboardForm
-      This is the form class used to render from the ModuleComponentFormView
+    """Abstract module related dashboard component based on forms.
+
+    This abstract component is used to make components from forms inheriting
+    from ModuleDashboardForm.
+    It allows to simply pass the form and some additional configuration
+    properties to create a valid component.
+    The components are registering urls based on the the unique identifier
+    and are using ModuleComponentFormView to handle the form processing.
+    Forms inheriting from ModuleDashboardForm may be configured with an
+    additional Meta.required attribute which defines the form fields that have
+    to be validly entered prior to project publishing. The component provides
+    a get_progress implementation based on this list and prevents from removing
+    required data if the project is published.
+
+    Properties
+    ----------
+    menu_label : str
+        This label is always returned regardless of module state
+    form_title : str
+        This title is shown on top of the rendered form
+    form_class : ModuleDashboardForm
```

Comment:
The type is wrong here: `form_class` should not be an instance of `ModuleDashboardForm` but a descendant